### PR TITLE
Add Calendar and Due Date to Tasks

### PR DESCRIPTION
**Pull request by ectherrien**
Adds ability to add a due date to all tasks. By default due date is todays date. There is logic that does not let you set a due date in the past. If you want to change the date just click on the date and it expands a calendar which allows you to select the new date. When done the Add button minimizes the calendar again.

Test Instructions:
1. Checkout the 'cal' branch and pull the latest changes for it
2. npm install
3. npm start
4. Make a new task
5. Hit 'View' on the new task, and todays date should be displayed as the due date
6. Make a new task, but this time click the current date and try to choose a date in the past (it should not let you), then choose a date in the future (remember this date)
7. Hit 'View' on the new task, and the date you chose in step 6 should e displayed as the due date

### CODE REVIEW

File: .npmignore
Comment (ectherrien) [inline .npmignore:1]: For some odd reason our project does not run unless I have this empty file...no idea why. If it causes anyone else issues we can add it to the .gitignore so it only exists on my computer and I do not push it up.

File: src/ListComponent.js [hunk 1]
```diff
@@ -7,22 +7,28 @@ import Button from "react-bootstrap/Button";
...
 import ListGroup from "react-bootstrap/ListGroup"; 
+import Calendar from 'react-calendar';
 import Popup from 'reactjs-popup';
 import 'reactjs-popup/dist/index.css';
 
 const ListComponent = (props) => { 
     const [input, setInput] = useState('');
     const [list, setList] = useState([]);
+    const [showCal, setShowCal] = useState(false);
+    const [calDate, setCalDate] = useState(new Date());
```
Comment (ectherrien) [inline src/ListComponent.js:19]: This is just temporary storage of the input from calendar, date is stored per task, and this is reset once a task is added.

File: src/ListComponent.js [hunk 2]
```diff
@@ -7,22 +7,28 @@ import Button from "react-bootstrap/Button";
...
         if (input !== "") { 
             const taskToAdd = { 
                 id: Math.random(),
                 title: input,
+                date: calDate,
                 details: ''
             };
             setList([...list, taskToAdd]);
             setInput('');
+            setShowCal(false);
```
Comment (ectherrien) [inline src/ListComponent.js:32]: Hides calendar when new task is added.

File: src/ListComponent.js [hunk 3]
```diff
@@ -7,22 +7,28 @@ import Button from "react-bootstrap/Button";
...
+            setCalDate(new Date());
```
Comment (ectherrien) [inline src/ListComponent.js:33]: Resets the due date of the next task to current date.

**Review by pragatipant — APPROVED** [review]
Changes look good to me. I tested it out as well. I think there might be a conflict that needs to be resolved based on the latest merged PR.

**Comment by pragatipant** [comment]
Looks like I was able to resolve the conflict myself, so merging this PR now.

**Pull request merged**